DDDigi/io/DigiIO.cpp

```diff
@@ -171,8 +171,8 @@ namespace dd4hep {
       mcp.setCharge(3.0*p.charge);
       mcp.setVertex( _toVectorD(p.start_position) );
       mcp.setEndpoint( _toVectorD(p.end_position) );
-      mcp.setMomentum( _toVectorD(p.momentum) );
-      mcp.setMomentumAtEndpoint( _toVectorD(p.momentum) );
+      mcp.setMomentum( _toVectorF(p.momentum) );
+      mcp.setMomentumAtEndpoint( _toVectorF(p.momentum) );
```

Comment:
I would assume that we run into [this deprecation warning](https://github.com/key4hep/EDM4hep/blob/285a6f04a47937b8769dd4495100f7c5507b5dcd/edm4hep.yaml#L71-L72) here with this construct? Similar below.

Reply:
Yes

Reply:
If we want to avoid pre-processor special casing we could change the definition of `_toVectorF` to

```cpp
template<typename T = edm4hep::Vector3f>
T _toVectorF(const dd4hep::Position& ep) {
      using FT = decltype(T::x);
      return { FT(ep.x()), FT(ep.y()), FT(ep.z()) };
} 
```
and then use it here as
```cpp
mcp.setMomentum( _toVectorF<(decltype(mcp.getMomentum())>(p.momentum) );
```

But a pre-processor `#if` definitely sticks out a bit more for a potential future cleanup.

Reply:
Went for the preprocessor check to be able to grep `EDM4HEP_VERSION`